### DISCUSSION TITLE
Lightweight persistence on disk (alternative to the beast that is CoreData) dependency?

### DISCUSSION THREAD

**Discussion by Sajjon** [tgrapperon/swift-dependencies-additions#44] [General]
Hey Thomas! Any plans on adding a lightweight persistence dependency that saves data (`Codable`s) to the file system? Potentially just a wrapper around some other library? I've used [Disk](https://github.com/saoudrizwan/Disk) in the past, but it seems dead, and is iOS only (was gonna fix that, but saw PRs seem to be ignored, because not maintained). So I took a look around and potentially these are candidates more suitable than Disk?
- [EasyStash](https://github.com/onmyway133/EasyStash)
- [Bodega](https://github.com/mergesort/Bodega)
- [Stores](https://github.com/omaralbeik/Stores) (also supports Keychain
- [Carlos](https://github.com/spring-media/Carlos)
- [Shallows](https://github.com/dreymonde/Shallows) (stated to be inspired by Carlos above)
- [Boutique](https://github.com/mergesort/Boutique)

Or are you reluctant adding a SPM dependency to implement a... dependency 🤪, I completely understand that if that is the case. If so, any plans for implementing this yourself?

**Comment by tgrapperon** [comment]
Hey @Sajjon! Thanks for opening the discussion.

The purpose of the library is to feel the most "built-in" and the less opinionated possible, and implementing such feature would depart from this.

The library ships with a dependency for `Codable`, and another one to write `Data` on disk, so it can serve as a basis to build your own abstraction on top of that. Or you can use the proxy tools to build your own wrapper of any of the libraries you listed. You can also wrap these libraries without using such tools of course.

I'm converting this issue to a discussion as this is not a direct issue with the library.
